zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/store/ElasticsearchBatchRequest.java

```diff
@@ -188,14 +191,29 @@ private void validateNoErrors(
       return;
     }
 
-    final String errorMessages =
+    final Map<ErrorKey, List<String>> groupedErrors =
         errorItems.stream()
+            .collect(
+                Collectors.groupingBy(
+                    item -> new ErrorKey(item.operationType(), item.index(), item.error().reason()),
+                    LinkedHashMap::new,
+                    Collectors.mapping(BulkResponseItem::id, Collectors.toList())));
```

Comment:
You could merge this with the `items.stream()...` on line 189, as it iterates through the same list